src/core/ComboBox/ComboBox.tsx

```diff
@@ -380,84 +459,365 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
+      </span>
+    );
+  },
+);
+ComboBoxEndIcon.displayName = 'ComboBoxEndIcon';
+
+const ComboBoxInput = React.forwardRef(
+  (props: InputProps, forwardedRef: React.RefObject<HTMLInputElement>) => {
+    const { onKeyDown: onKeyDownProp, onFocus: onFocusProp, ...rest } = props;
+
+    const { isOpen, id, focusedIndex } = useSafeContext(ComboBoxStateContext);
+    const dispatch = useSafeContext(ComboBoxActionContext);
+    const { inputRef, menuRef, optionsRef } = useSafeContext(
+      ComboBoxRefsContext,
+    );
+    const refs = useMergedRefs(inputRef, forwardedRef);
+
+    const focusedIndexRef = React.useRef({ original: -1, filtered: -1 });
+
+    // FIXME: Currently, the selected item is not focused on the first render.
+
+    // const forceRerender = React.useReducer(() => ({}), {})[1];
+    // const [optionsMounted, setOptionsMounted] = React.useState(false);
+    // React.useEffect(() => {
+
+    //   if (!isOpen) {
+    //     return;
+    //   }
+
+    //   if (isOpen && !menuRef.current) {
+    //     forceRerender();
+    //   }
+
+    //   if (menuRef.current?.children.length && optionsRef.current.length > 0) {
+    //     setOptionsMounted(true);
+    //   }
+    // });
+
+    React.useEffect(() => {
+      focusedIndexRef.current = {
+        original: focusedIndex ?? -1,
+        filtered: optionsRef.current.findIndex(
+          (el) => Number(el.dataset.iuiIndex) === focusedIndex,
+        ),
+      };
+    }, [focusedIndex, optionsRef]);
+
+    const getIdFromIndex = (index: number) => {
+      return (
+        menuRef.current?.querySelector(`[data-iui-index="${index}"]`)?.id ?? ''
+      );
+    };
+
+    const handleKeyDown = React.useCallback(
+      (event: React.KeyboardEvent<HTMLInputElement>) => {
+        onKeyDownProp?.(event);
+        const length = optionsRef.current.length ?? 0;
+
+        switch (event.key) {
+          case 'ArrowDown': {
+            event.preventDefault();
+            if (!isOpen) {
+              return dispatch(['open']);
+            }
+
+            if (focusedIndexRef.current.original === -1 && length > 0) {
+              return dispatch([
+                'focus',
+                Number(optionsRef.current[0].dataset.iuiIndex),
+              ]);
+            }
+
+            if (length === 0) {
+              return;
             }
-          }}
-          animation='shift-away'
-          duration={200}
-          {...dropdownMenuProps}
-          content={
-            <Menu
-              id={`${id}-list`}
-              className='iui-scroll'
-              style={{
-                minWidth,
-                maxWidth: `min(${minWidth * 2}px, 90vw)`,
-                maxHeight: 300,
-              }}
-              setFocus={false}
-              role='listbox'
-              ref={menuRef}
-            >
-              {menuItems}
-            </Menu>
+
+            let nextIndex = focusedIndexRef.current.filtered;
+            do {
```

Comment:
Can you just get all focusable items and increase index of focusable list?

Reply:
getFocusableElements will not work with virtualized lists because the elements are not always in the DOM